server/src/main/java/org/elasticsearch/search/dfs/DfsPhase.java

```diff
@@ -39,6 +45,11 @@ public class DfsPhase {
 
     public void execute(SearchContext context) {
         try {
+            final boolean[] profile = new boolean[2];
```

Comment:
I agree that a comment is needed. I can probably rework this to just a single boolean after thinking about it. The idea is to avoid tracking time for createWeight used for collecting term stats, and only turning on profiling for createWeight when run the actual vector query.

Reply:
@nik9000 maybe it would make sense to create a new `IndexSearcher` that wraps the context reader specifically for knn search instead of re-using the one designed for term stats? I just wasn't sure how heavy-weight it is to make a second one? I think having a separate one for term stats and a separate one for knn may make sense though? Probably also separate out the terms stats collection and knn search into separate private methods?

Reply:
Yeah. I'm not really sure the "right" way to fix it, but I'll bet there is some nicer way.....

Reply:
@nik9000 I updated this section to split out the dfs phase into two private methods where one does stats collection and the other does the knn search. Each method creates its own `IndexSearcher` to wrap the existing `IndexReader`. This allows us to override methods only relevant to profiling the knn search in what I consider more obvious way and eliminates the `boolean[]` hack. What do you think?